### PR TITLE
Example module mod-spring-petstore added

### PR DESCRIPTION
**Pull request by TarasSpashchenko**
This is an example of the FOLIO backend module built using folio-spring-base library.

The module has been created using the mod-spring-template.

### CODE REVIEW

**Review by ncovercash — APPROVED** [review]

File: mod-spring-petstore/descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      "displayName": "pets module - all permissions",
+      "description": "Entire set of permissions needed to use the pets module",
+      "subPermissions": [
+        "pets.collection.get",
+        "pets.item.get",
+        "pets.item.post"
+      ],
+      "visible": false
+    }
+  ],
```
Comment (dcrossleyau) [inline mod-spring-petstore/descriptors/ModuleDescriptor-template.json:83]: jq reports a JSON error

File: mod-spring-petstore/descriptors/DeploymentDescriptor-template.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "srvcId": "@artifactId@-@version@",
+  "nodeId": "localhost",
+  "descriptor": {
+    "exec": "java -Dport=%p -jar ../mod-spring-template/target/mod-spring-petstore.jar -Dhttp.port=%p embed_postgres=true"
```
Comment (dcrossleyau) [inline mod-spring-petstore/descriptors/DeploymentDescriptor-template.json:5]: Is it possible to use the replacement tokens here too.

File: mod-spring-petstore/descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    "env": [
+      { "name": "JAVA_OPTIONS",
+        "value": "-XX:MaxRAMPercenpete=66.0"
+      },
+      { "name": "DB_HOST", "value": "postgres" },
+      { "name": "DB_PORT", "value": "5432" },
+      { "name": "DB_USERNAME", "value": "TODO" },
+      { "name": "DB_PASSWORD", "value": "TODO" },
+      { "name": "DB_DATABASE", "value": "TODO" },
```
Comment (dcrossleyau) [inline mod-spring-petstore/descriptors/ModuleDescriptor-template.json:102]: The actual values are provided: https://dev.folio.org/guides/module-descriptor/#env-db-environment

**Review by dcrossleyau — CHANGES_REQUESTED** [review]
As noted in today's [mod-spring-template/pull/9](https://github.com/folio-org/mod-spring-template/pull/9) this should ideally be using Java 17.

https://dev.folio.org/faqs/how-to-specify-backend-java-ci/

Should then also use 'wget' for the "health check" rather than 'curl' [FOLIO-3507](https://issues.folio.org/browse/FOLIO-3507).

File: mod-spring-petstore/descriptors/DeploymentDescriptor-template.json (hunk 2 above)
Comment (TarasSpashchenko) [reply mod-spring-petstore/descriptors/DeploymentDescriptor-template.json:5]: Could you please explain a bit about that? What should be replaced?

Comment (dcrossleyau) [reply mod-spring-petstore/descriptors/DeploymentDescriptor-template.json:5]: "mod-spring-petstore" with "@artifactId@"

Also "mod-spring-template" is incorrect, and perhaps could use that Maven replacement token too.

Comment (dcrossleyau) [reply mod-spring-petstore/descriptors/DeploymentDescriptor-template.json:5]: IIUC then "embed_postgres" is not used anymore.

File: mod-spring-petstore/src/main/resources/banner.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ -dddd.    :dddd`    sdddy   dddd:   odddy  .dddd:    :dddd.   .++.
+ -dddd.    hdddy     /dddd:  dddd:   odddy  odddd`    `ddddo
+ -dddd.    ddddy     :dddd/  dddd:   odddy  sdddd      dddds
+ -dddd.    ydddh     /dddd-  dddd:   odddy  +dddd`    `dddd+
+ -dddd.    -dddd.    ydddo   dddd:   odddy  `yddd+    /dddh`
+ -dddd.     .sdddo++yddh:    dddd:   odddy   `oddds++sddd+`
+ `////`       `:/+o++:.      ////.   -///:      -/+oo+/-
+
+  (${application.title} ${application.version} with Spring Boot${spring-boot.formatted-version})
+-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```
Comment (dmtkachenko) [inline mod-spring-petstore/src/main/resources/banner.txt:19]: 👍 😄 where have you been with this banner before??

File: mod-spring-petstore/src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        <Console name="Console" target="SYSTEM_OUT">
+            <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+        </Console>
+    </Appenders>
+    <Loggers>
+        <Root level="info">
+            <AppenderRef ref="Console"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (dmtkachenko) [inline mod-spring-petstore/src/main/resources/log4j2.xml:13]: standard log4j configuration comes from folio-spring-base.
modules should not define custom settings

File: mod-spring-petstore/src/main/resources/banner.txt (hunk 4 above)
Comment (ncovercash) [reply mod-spring-petstore/src/main/resources/banner.txt:19]: @dmtkachenko that art was me!  Got tired of staring at the default Spring Boot one, lol

File: mod-spring-petstore/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,20 @@
+buildMvn {
+  publishModDescriptor = true
+  mvnDeploy = true
+  doKubeDeploy = true
+  buildNode = 'jenkins-agent-java11'
```
Comment (dcrossleyau) [inline mod-spring-petstore/Jenkinsfile:5]: Should be 'jenkins-agent-java17'
https://dev.folio.org/faqs/how-to-specify-backend-java-ci/

File: mod-spring-petstore/descriptors/DeploymentDescriptor-template.json (hunk 2 above)
Comment (TarasSpashchenko) [reply mod-spring-petstore/descriptors/DeploymentDescriptor-template.json:5]: Done

File: mod-spring-petstore/src/main/resources/log4j2.xml (hunk 5 above)
Comment (TarasSpashchenko) [reply mod-spring-petstore/src/main/resources/log4j2.xml:13]: Done

File: mod-spring-petstore/Jenkinsfile (hunk 6 above)
Comment (TarasSpashchenko) [reply mod-spring-petstore/Jenkinsfile:5]: Done

File: mod-spring-petstore/descriptors/ModuleDescriptor-template.json (hunk 3 above)
Comment (TarasSpashchenko) [reply mod-spring-petstore/descriptors/ModuleDescriptor-template.json]: Done

**Review by dcrossleyau — APPROVED** [review]

**Pull request merged**